### ISSUE TITLE
Trait doku::Document is not implemented, but example string is given

### ISSUE THREAD

**Issue by Nutomic** [anixe/doku#19]
I have the following field:
```
  #[default(Url::parse("http://pictrs:8080").unwrap())]
  #[doku(example = "http://pictrs:8080")]
  pub url: Url,
```

It fails to comopile with `the trait doku::Document is not implemented for url::Url`. I would expect this to work, by doku using the explicitly provided example, which is already a string. Alternatively, Url implements ToString, so that could also be used with the default value.

**Comment by Patryk27** [comment]
Hopefully [specialization](https://github.com/rust-lang/rust/issues/31844) will allow Doku to be smarter around cases such as these, but currently even when using `example = ...`, the underlying type must implement Doku's `Document` trait.

Fortunately, adding support for those is a breeze! 

tl;dr I've just pushed a commit that makes Doku able to document URLs, which you can activate using:

```toml
doku = { git = "https://github.com/anixe/doku", features = ["url-2"] }
```
(note that I assumed you're using https://crates.io/crates/url, version 2.x.x - if that's some other crate or an older version, just lemme know.)

If that works for you, I should be able to deploy the updated version to crates.io soon 🙂

**Comment by Nutomic** [comment]
Thanks for the quick fix! It almost works, but there is a strange error which i dont understand:
```
53  |     println!("{}", doku::to_json_fmt_val(&fmt, &Settings::default()));
|                    ---------------------       ^^^^^^^^^^^^^^^^^^^^ the trait `Document` is not implemented for `Settings`
|                    |
|                    required by a bound introduced by this call
```

Code is on this branch: https://github.com/LemmyNet/lemmy/pull/2343

**Comment by Nutomic** [comment]
Ah thanks, i didnt notice that there was a second doku dep. It works fine now, so you can probably merge.

**Comment by Patryk27** [comment]
Sorry for late response - newer Doku has been deployed! 🙂 

So:
```toml
doku = { version = "0.12.0", features = ["url-2"] }
```

**Comment by Nutomic** [comment]
No problem, thanks!
